src/constraints/BigIntConstraints.ts

```diff
@@ -1,56 +1,44 @@
-import { ConstraintError, ConstraintErrorMessageBuilder } from '../lib/errors/ConstraintError';
+import { ConstraintError } from '../lib/errors/ConstraintError';
 import { Result } from '../lib/Result';
 import type { IConstraint } from './base/IConstraint';
 import { Comparator, eq, ge, gt, le, lt, ne } from './util/operators';
 
-function bigintComparator(
-	comparator: Comparator,
-	name: string,
-	messageBuilder: ConstraintErrorMessageBuilder<bigint>,
-	number: bigint
-): IConstraint<bigint> {
+function bigintComparator(comparator: Comparator, name: string, expected: string, number: bigint): IConstraint<bigint> {
```

Comment:
Is it possible to make these comparator `name` parameters strictly typed?

I'd prefer if you retroactively added a type along the lines of `BigIntComparatorNames = 's.bigint.lt' | 's.bigint.le'` etc.

That way this code is less prone to breaking when we revisit it 6 months from now. Same goes for all of these such array and all that. Alternatively if you have another way to type it strictly that's fine too, just as long as where we call the comparators we have strict typing and autocomplete. For example an alternative might be having one shared interface for all comparators and accessing the string union for this specific one with `SharedInterface['bigint']`

Reply:
Addressed that in 41bebea.